packages/glimmer-runtime/lib/compiled/opcodes/dom.ts

```diff
@@ -221,6 +223,85 @@ export class StaticAttrOpcode extends Opcode {
   }
 }
 
+export class ModifierOpcode extends Opcode {
+  public type = "modifier";
+  public name: InternedString;
+  public args: CompiledArgs;
+  private manager: ModifierManager<Opaque>;
+
+  constructor({ name, manager, args }: { name: InternedString, manager: ModifierManager<Opaque>, args: CompiledArgs }) {
+    super();
+    this.name = name;
+    this.manager = manager;
+    this.args = args;
+  }
+
+  evaluate(vm: VM) {
+    let { manager } = this;
+    let stack = vm.stack();
+    let { element } = stack;
+    let args = this.args.evaluate(vm);
+
+    let modifier = manager.install(element, vm.frame.getArgs());
+    let destructor = manager.getDestructor(modifier);
+
+    if (destructor) {
+      vm.newDestroyable(destructor);
+    }
+
+    vm.updateWith(new UpdateModifierOpcode({
+      manager,
+      modifier,
+      element,
+      args
+    }));
+  }
+
+  toJSON(): OpcodeJSON {
+    let { _guid: guid, type, name, args } = this;
+
+    let details = dict<string>();
+
+    details["type"] = JSON.stringify(type);
+    details["name"] = JSON.stringify(name);
+    details["args"] = JSON.stringify(args);
+
+    return { guid, type, details };
+  }
+}
+
+export class UpdateModifierOpcode extends DOMUpdatingOpcode {
+  public type = "update-modifier";
+
+  private element: Element;
+  public args: EvaluatedArgs;
+  private manager: ModifierManager<Opaque>;
+  private modifier: Opaque;
+
+  constructor({ manager, modifier, element, args }: { manager: ModifierManager<Opaque>, modifier: Opaque, element: Element, args: EvaluatedArgs }) {
+    super();
+    this.modifier = modifier;
+    this.manager = manager;
+    this.element = element;
+    this.args = args;
+  }
+
+  evaluate(vm: UpdatingVM) {
+    // Need to:
+    // 1) Check if args has changed (need a cache?)
+    let { manager, modifier, element, args } = this;
+    manager.update(modifier, element, args);
```

Comment:
You can skip that for now (just always call update - as long as your tests aren't asserting how many times that get called it shouldn't block you). I can show you the code later 👍